tests/integration/index.ts

```diff
@@ -186,81 +214,79 @@ describe("Positive Cases", () => {
       // //////////
       // // MINT //
       // //////////
-      await expect(vBNX.connect(acc2Signer).mint(mintAmount))
-        .to.emit(vBNX, "Mint")
-        .withArgs(acc2, mintAmount, mintAmount, mintAmount);
+      const tx = await vBNX.connect(acc2Signer).mint(mintAmount);
+      await expect(tx).to.emit(vBNX, "Mint").withArgs(acc2, mintAmount, vTokenMintAmount, vTokenMintAmount);
       [error, balance, borrowBalance] = await vBNX.connect(acc2Signer).getAccountSnapshot(acc2);
       expect(error).to.equal(Error.NO_ERROR);
-      expect(balance).to.equal(mintAmount);
+      expect(balance).to.equal(vTokenMintAmount);
       expect(borrowBalance).to.equal(0);
       [error, liquidity, shortfall] = await Comptroller.connect(acc2Signer).getAccountLiquidity(acc2);
       expect(error).to.equal(Error.NO_ERROR);
-      expect(liquidity).to.equal(new BigNumber(mintAmount * collateralFactor).multipliedBy(vBNXPrice));
+      expect(liquidity).to.equal(new BigNumber(mintAmount).multipliedBy(collateralFactor).multipliedBy(vBNXPrice));
       expect(shortfall).to.equal(0);
       // ////////////
       // // Borrow //
       // ////////////
       // Supply WBTC to market from 2nd account
       await expect(vBSW.connect(acc1Signer).mint(mintAmount))
         .to.emit(vBSW, "Mint")
-        .withArgs(await acc1Signer.getAddress(), mintAmount, mintAmount, mintAmount);
-
+        .withArgs(await acc1Signer.getAddress(), mintAmount, vTokenMintAmount, vTokenMintAmount);
       [error, balance, borrowBalance] = await vBSW
         .connect(acc2Signer)
         .getAccountSnapshot(await acc1Signer.getAddress());
       expect(error).to.equal(Error.NO_ERROR);
-      expect(balance).to.equal(mintAmount);
+      expect(balance).to.equal(vTokenMintAmount);
       expect(borrowBalance).to.equal(0);
-
       [error, liquidity, shortfall] = await Comptroller.connect(acc1Signer).getAccountLiquidity(acc1);
       expect(error).to.equal(Error.NO_ERROR);
-      expect(liquidity).to.equal(new BigNumber(mintAmount * collateralFactor).multipliedBy(vBSWPrice));
+      expect(liquidity).to.equal(new BigNumber(mintAmount).multipliedBy(collateralFactor).multipliedBy(vBSWPrice));
       expect(shortfall).to.equal(0);
-
-      const bswBorrowAmount = 1e4;
+      const bswBorrowAmount = convertToUnit(1, 18);
+      const vBSWBorrowAmount = convertToUnit(1, 18);
 
       await expect(vBSW.connect(acc2Signer).borrow(bswBorrowAmount))
         .to.emit(vBSW, "Borrow")
-        .withArgs(acc2, bswBorrowAmount, bswBorrowAmount, bswBorrowAmount);
-
+        .withArgs(acc2, bswBorrowAmount, vBSWBorrowAmount, vBSWBorrowAmount);
```

Comment:
The third parameter is `accountBorrowsNew` and the last one `totalBorrowsNew`, so in both cases they are referencing the underlying token.
```suggestion
        .withArgs(acc2, bswBorrowAmount, bswBorrowAmount, bswBorrowAmount);
```

Reply:
Same here

Reply:
Fixed